build.gradle

```diff
@@ -26,15 +28,66 @@ jar.manifest {
 
 buildscript {
     repositories {
+        mavenCentral()
+        mavenLocal()
         maven { url "https://plugins.gradle.org/m2/" }
     }
+    dependencies {
+        classpath 'net.ltgt.gradle:gradle-errorprone-plugin:0.0.13'
+        classpath "gradle.plugin.com.github.sherter.google-java-format:google-java-format-gradle-plugin:0.7.1"
+    }
 }
 
 def opencensusVersion = '0.16.1'
+def errorProneVersion = '2.3.1'
 
 dependencies {
     compile "io.opencensus:opencensus-api:${opencensusVersion}"
     runtime "io.opencensus:opencensus-impl:${opencensusVersion}"
+
+    compileOnly "com.google.errorprone:error_prone_annotations:${errorProneVersion}"
+}
+
+compileJava {
+    // We suppress the "try" warning because it disallows managing an auto-closeable with
+    // try-with-resources without referencing the auto-closeable within the try block.
+    // We suppress the "processing" warning as suggested in
+    // https://groups.google.com/forum/#!topic/bazel-discuss/_R3A9TJSoPM
+    it.options.compilerArgs += ["-Xlint:all", "-Xlint:-try", "-Xlint:-processing"]
+    it.options.compilerArgs += ["-XepAllDisabledChecksAsWarnings", "-XepDisableWarningsInGeneratedCode"]
+
+    // MutableMethodReturnType can suggest returning Guava types from
+    // API methods (https://github.com/google/error-prone/issues/982).
+    it.options.compilerArgs += ["-Xep:MutableMethodReturnType:OFF"]
+
+    // ReturnMissingNullable conflicts with Checker Framework null analysis.
+    it.options.compilerArgs += ["-Xep:ReturnMissingNullable:OFF"]
+
+    // OpenCensus doesn't currently use Var annotations.
+    it.options.compilerArgs += ["-Xep:Var:OFF"]
+
+    it.options.encoding = "UTF-8"
+    it.options.compilerArgs += ["-Xlint:-cast"]
+    it.options.compilerArgs += ["-Werror"]
+}
+
+// Google formatter works only on java8.
+if (JavaVersion.current().isJava8Compatible()) {
+    googleJavaFormat {
+        toolVersion '1.6'
```

Comment:
What's '1.6' if the comment above says it works only 1.8?

Reply:
`1.6` is the version for Google Java Formatter, while `1.8` is the JDK version.